frontend/components/form-spinner/form-spinner.js

```diff
@@ -0,0 +1,32 @@
+import { nodeListForEach } from '../../utils'
+
+function FormSpinner(container) {
+  this.container = container
+
+  this.spinnerSvg =
+    '<div class="lds-spinner"><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div>'
+
+  this.container.addEventListener('submit', () => {
+    const formSpinnerTemplate = document.createElement('template')
+    formSpinnerTemplate.innerHTML = `
+      <div class="form-spinner">
+        <div class="form-spinner__notification-box" role="alert">
+          Loading
+          <div class="form-spinner__spinner">
+            ${this.spinnerSvg}
+          </div>
+        </div>
+      </div>
+    `.trim()
+
+    document.querySelector('body').appendChild(formSpinnerTemplate.content.firstChild)
+
+    const buttons = this.container.querySelectorAll('[data-module="govuk-button"]')
+    nodeListForEach(buttons, function (button) {
+      button.setAttribute('disabled', 'disabled')
+      button.setAttribute('aria-disabled', 'true')
+    })
```

Comment:
The overlay will make elements unclickable, however keyboard users can still tab to elements below the overlay. This disables any submit buttons on the form to prevent double submissions via keyboard.